crates/bevy_core_pipeline/src/dof/mod.rs

```diff
@@ -0,0 +1,907 @@
+//! of field effect in Bevy is based on physical parameters.
+//!
+//! [Depth of field]: https://en.wikipedia.org/wiki/Depth_of_field
+
+use std::f32::INFINITY;
+
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, Handle};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    query::{QueryItem, With},
+    schedule::IntoSystemConfigs as _,
+    system::{lifetimeless::Read, Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_render::{
+    camera::{PhysicalCameraParameters, Projection},
+    extract_component::{ComponentUniforms, DynamicUniformIndex, UniformComponentPlugin},
+    render_graph::{
+        NodeRunError, RenderGraphApp as _, RenderGraphContext, ViewNode, ViewNodeRunner,
+    },
+    render_resource::{
+        binding_types::{
+            sampler, texture_2d, texture_depth_2d, texture_depth_2d_multisampled, uniform_buffer,
+        },
+        BindGroup, BindGroupEntries, BindGroupLayout, BindGroupLayoutEntries,
+        CachedRenderPipelineId, ColorTargetState, ColorWrites, FilterMode, FragmentState, LoadOp,
+        Operations, PipelineCache, RenderPassColorAttachment, RenderPassDescriptor,
+        RenderPipelineDescriptor, Sampler, SamplerBindingType, SamplerDescriptor, Shader,
+        ShaderStages, ShaderType, SpecializedRenderPipeline, SpecializedRenderPipelines, StoreOp,
+        TextureDescriptor, TextureDimension, TextureFormat, TextureSampleType, TextureUsages,
+    },
+    renderer::{RenderContext, RenderDevice},
+    texture::{BevyDefault, CachedTexture, TextureCache},
+    view::{
+        prepare_view_targets, ExtractedView, Msaa, ViewDepthTexture, ViewTarget, ViewUniform,
+        ViewUniformOffset, ViewUniforms,
+    },
+    Extract, ExtractSchedule, Render, RenderApp, RenderSet,
+};
+use bevy_utils::{info_once, prelude::default, warn_once};
+use smallvec::SmallVec;
+
+use crate::{
+    core_3d::{
+        graph::{Core3d, Node3d},
+        Camera3d,
+    },
+    fullscreen_vertex_shader::fullscreen_shader_vertex_state,
+};
+
+const DOF_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(2031861180739216043);
+
+/// A plugin that adds support for the depth of field effect to Bevy.
+pub struct DepthOfFieldPlugin;
+
+/// Depth of field settings.
+#[derive(Component, Clone, Copy)]
+pub struct DepthOfFieldSettings {
+    /// The appearance of the effect.
+    pub mode: DepthOfFieldMode,
+
+    /// The distance in meters to the location in focus.
+    pub focal_distance: f32,
+
+    /// The height of the [image sensor format] in meters.
+    ///
+    /// Focal length is derived from the FOV and this value. The default is
+    /// 18.66mm, matching the [Super 35] format, which is popular in cinema.
+    ///
+    /// [image sensor format]: https://en.wikipedia.org/wiki/Image_sensor_format
+    ///
+    /// [Super 35]: https://en.wikipedia.org/wiki/Super_35
+    pub sensor_height: f32,
```

Comment:
A user might not know what changing the sensor height might do to depth of field, could use a bit more clarification